### PR TITLE
Members only project using devise

### PR DESCRIPTION
**Pull request by arikarim**
In this project, we implemented the basic use of authentication using 'devise' Gem.

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Nihal Mohammed & Ari Ali Hars** 
Good job with your first submission for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a couple of details to improve before you can go ahead:

- [x] After the user signs in or sign up it should be able to see a link (or button) to the posts/new page.
- [x] "sign out" button should also be available in posts/new page.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Comment by arikarim** [comment]
thanks for the review, we did all the changes required.

please have a look.

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        
+<% end %>
+
+<% if user_signed_in? %>
+    <%= button_to 'sign out', destroy_user_session_path, method: :delete %>
+    <%= link_to 'new post', new_post_path, method: :delete %>
+<% else %>
+    <%= button_to 'sign in', new_user_session_path %>
+    <%= link_to 'sign up', new_user_registration_path %>
+<% end %>
```
Comment (acushlakoncept) [inline app/views/posts/index.html.erb]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @arikarim and @apricot12 , Great job! 👏  Both of you have done well on this project.

Let's make a few changes, and we will be done.

I am sorry for bringing this up, but I am sure it will help you in your future projects 

- [x] Your commit messages should be capitalized. You can watch this [video](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6) on how to edit your commit messages using git rebase
 See [Git Commit Best Practices](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) for more insights

- [x]  There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

Do this and you should be good to go. 🙏 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by peterrobert — APPROVED** [review]
### APPROVED 👍🏿 💯 
Hello, @arikarim 👋🏿 . Very nice project. Everything is working as expected. Great work creating the helpers for your logic 👏🏿 

### ON TO THE NEXT PROJECT 👍🏿 

### HAPPY CODING 💻

**Pull request merged**